File: 백준/Silver/11399. ATM/ATM.java
import java.util.*;
import java.io.*;

public class Main {

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		int[] times = new int[N];
		
		StringTokenizer st = new StringTokenizer(br.readLine());
		for(int i = 0; i < N; i++) {
			times[i] = Integer.parseInt(st.nextToken());
		}
		
		Arrays.sort(times);
		
		int sum = 0;
		int total = 0;
		for(int time: times) {
			sum += time;
			total += sum;
		}
		
		System.out.println(total);
	}

}

File: 백준/Silver/11399. ATM/ATM.kt
package org.example

import java.util.StringTokenizer

fun main() {
    val br = System.`in`.bufferedReader()
    val N = br.readLine().toInt()
    val st = StringTokenizer(br.readLine())
    val arr = mutableListOf<Int>()
    repeat(N){
        arr.add(st.nextToken().toInt())
    }
    arr.sort()
    var result = 0
    //println(arr)
    for(i in 0 until N){
        result += arr[i]*(N-i)
    }
    println(result)
    br.close()
}